wrapper/src/main/java/software/amazon/jdbc/HostSelector.java

```diff
@@ -18,17 +18,19 @@
 
 import java.sql.SQLException;
 import java.util.List;
+import java.util.Properties;
 
 public interface HostSelector {
 
   /**
    * Selects a host with the requested role from the given host list.
    *
-   * @param hosts a list of available hosts to pick from
-   * @param role  the desired host role - either a writer or a reader
+   * @param hosts a list of available hosts to pick from.
+   * @param role  the desired host role - either a writer or a reader.
+   * @param props connection properties that may be needed by the host selector in order to choose a host.
    * @return a host matching the requested role
    * @throws SQLException if the host list does not contain any hosts matching the requested role or
    *                      an error occurs while selecting a host
    */
-  HostSpec getHost(List<HostSpec> hosts, HostRole role) throws SQLException;
+  HostSpec getHost(List<HostSpec> hosts, HostRole role, Properties props) throws SQLException;
```

Comment:
Should we annotate props with @Nullable? It seems that some implementation may not need properties.